Add offline mode indicator

gambinish opened the pull request:
To run locally and test in browser:

1. Be sure to close any open `localhost:3000` ports/tabs

2. Then:

`npm run build`
`npm run serve`

3. Open up a new tab at `localhost:3000`
4. Open network tab, and toggle between offline, no throttling, 3g, etc.

Expected behavior: Green Alert when online, Red alert when offline.

https://github.com/gambinish/uswds-cra-pwa-ts/assets/35090461/648fd49b-bbfb-4b5b-9564-c09becf80b2b

gambinish committed:
Add offline mode indicator

gambinish committed:
cleanup

brockinit commented on src/index.tsx:
```diff
@@ -1,23 +1,29 @@
 // unregister() to register() below. Note this comes with some pitfalls.
 // Learn more about service workers: https://cra.link/PWA
-serviceWorkerRegistration.unregister();
+serviceWorkerRegistration.register();
```
Why does the service worker need to be registered here?

gambinish replied:
I think this is required to let the app load offline. I just followed the comment that CRA left and looked at their docs. Seems like you need this for the PWA to function offline?

Admittedly I don't have a ton of experience with service workers, so if this is wrong I can undo the change

thgaskell commented on src/App.tsx:
```diff
@@ -8,36 +7,61 @@ import {
 } from "@trussworks/react-uswds";
+import { useQuery, useQueryClient } from "react-query";
+
+const ping = async () => {
```
This should be refactored into a useEffect hook that uses the https://developer.mozilla.org/en-US/docs/Web/API/Navigator/onLine API.

gambinish replied:
Sounds good! Thanks for the clarification. Will remove react-query since its not really needed here.

thgaskell reviewed (changes requested):
Please remove the ping method. We don't want to rely on a single server response to determine if the device's online status. The navigator API will let us know at the device's networking level whether it's connected or not.

gambinish committed:
Remove ping method

gambinish committed:
Remove react-query dependency

gambinish commented:
@thgaskell 

removed ping method in favor of navigator api here: [29a0344](https://github.com/gambinish/uswds-cra-pwa-ts/pull/1/commits/29a0344e6b3e28a7faf751cbd132f4deef793fab)

removed react-query (for now) here: [5fac13d](https://github.com/gambinish/uswds-cra-pwa-ts/pull/1/commits/5fac13db2b25bd3ec081a553cb0b062c6c22e39d)
